pulsar-common/src/main/java/org/apache/pulsar/common/policies/data/TopicStats.java

```diff
@@ -131,7 +136,12 @@ public TopicStats add(TopicStats stats) {
             }
         } else {
             for (String repl : stats.replication.keySet()) {
-                this.replication.get(repl).add(stats.replication.get(repl));
+                if (this.replication.get(repl) != null) {
```

Comment:
same comment above

Reply:
I am not sure. Neither the map itself nor the class SubscriptionStats looks thread-safe. Other fields like `publishers' are not prepared to concurrent updates neither.

Reply:
If you are already changing the map from a `Map` to a `ConcurrentMap`, doesn't it mean there are concurrent updates? If so, since we already are using a concurrent map, we should consider using CAS operations to do better, no?

Reply:
@sijie These are 2 fixes for 2 issues. The ConcurrentMap change is for another issue. By the way, I believe the other issue is indeed a threading problem, but I don't know if the current fix is good enough.

Reply:
@sijie Could you accept my WeChat request? The turn around time here is long.

Reply:
I am not convinced about the comment here. If the problem is a concurrent modification to a HashMap, changing a HashMap to a ConcurrentMap doesn't address the problem. We also need to change the way how we use a concurrent hash map. 

If there are concurrent accesses, line 143 can be accessed multiple times. There will be multiple replicator stats added to the map. A newer stats can replace a stats that might be already accessed by other threads. It results inconsistent behavior when accessing stats object.

Same comment about at line 139. You check if a replication exists in line 139. but you didn't use the replication stats you got in line 139. But In line 140, you try to get the replication stats again. If there are concurrent accesses, the replication might be removed at the time executing line 139 and line 140. This results in NullPointerException.

This change is actually introducing more problems if there are concurrent accesses. The code here should be rewritten as the followings:

```
ReplicatorStats oldStats = this.replication.get(repl);
if (oldStats == null) {
     ReplicatorStats newStats = new ReplicatorStats();
     oldStats = this.replication.putIfAbsent(repl, newStats);
     if (oldStats == null) {
          oldStats = newStats;
     }
}
oldStats.add(stats.replication.get(repl));
```